src/distribution.cpp

```diff
@@ -26,30 +27,69 @@ Discrete::Discrete(pugi::xml_node node)
   auto params = get_node_array<double>(node, "parameters");
 
   std::size_t n = params.size();
-  std::copy(params.begin(), params.begin() + n / 2, std::back_inserter(x_));
-  std::copy(params.begin() + n / 2, params.end(), std::back_inserter(p_));
+  double x[n / 2], p[n / 2];
 
-  normalize();
+  std::copy(params.begin(), params.begin() + n / 2, x);
+  std::copy(params.begin() + n / 2, params.end(), p);
+
+  new (this) Discrete(x, p, n / 2);
```

Comment:
Ok, sorry for the back and forth. Evidently this approach is [not advisable](https://isocpp.org/wiki/faq/ctors#init-methods). Let's revert back to having an `init` method but it should accept a vector of `x` and `p`. So, we'd have something like:
```c++
Discrete::Discrete(pugi::xml_node node)
{
  auto params = get_node_array<double>(node, "parameters");

  std::size_t n = params.size();
  std::vector<double> x(params.begin(), params.begin() + n / 2);
  std::vector<double> p(params.begin() + n / 2, params.end());

  this->init(x, p);
}

Discrete::Discrete(const double* x, const double* p, int n)
{
  std::vector<double> x_vec(x, x + n);
  std::vector<double> p_vec(p, p + n);
  this->init(x_vec, p_vec);
}

void Discrete::init(const std::vector<double>& x, const std::vector<double>& p)
{
  ...
}
```